crates/types/src/net/log_server.rs

```diff
@@ -593,3 +604,88 @@ impl TailUpdated {
         self
     }
 }
+
+// ** GET_DIGEST
+
+/// Request a digest of the loglet between two offsets from this node
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct GetDigest {
+    #[serde(flatten)]
+    pub header: LogServerRequestHeader,
+    // inclusive
+    pub from_offset: LogletOffset,
+    pub to_offset: LogletOffset,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, derive_more::Display, Serialize, Deserialize)]
+#[display("[{from_offset}..{to_offset}] -> {status} ({})",  self.len())]
+pub struct DigestEntry {
+    // inclusive
+    pub from_offset: LogletOffset,
+    pub to_offset: LogletOffset,
+    pub status: RecordStatus,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, derive_more::Display, Serialize, Deserialize)]
+#[repr(u8)]
+pub enum RecordStatus {
+    #[display("T")]
+    Trimmed,
+    #[display("A")]
+    Archived,
+    #[display("X")]
+    Exists,
+}
+
+impl DigestEntry {
+    // how many offsets are included in this entry
+    pub fn len(&self) -> usize {
+        debug_assert!(self.to_offset >= self.from_offset);
```

Comment:
What if the `DigestEntry` is empty and I want to call `len` on it? Wouldn't it be better to return 0 than panicking due to the assertion? If we are running w/o debug assertions, len would return 1 for an empty entry.

Reply:
Good call. Let me do that.